sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java

```diff
@@ -405,6 +477,8 @@ protected void hookOnError(Throwable throwable) {
             performWorkload(baseSubscriber, i);
         }
 
+        System.clearProperty("COSMOS.MIN_CONNECTION_POOL_SIZE_PER_ENDPOINT");
```

Comment:
Isn't this too late? Client has been created already?

Reply:
Being late here is intentional since I am clearing the system property. When setting the same system property, I'd like to be early - basically before my clients have been initialized.

Reply:
But why clear it at all - we are never reusing the process for another benchmark, do we? It is always run a benchmark, close the process - so why even attempt to clear the system property? Does that not make it harder to debug?

Reply:
Makes sense. No need to clear the property.